tests/e2e/topology_operation_strom_cases.go

```diff
@@ -0,0 +1,683 @@
+		14. Scale down the Statefulsets replica count to 5,
+		15. While the Statefulsets is scaling down Pods, kill CSI Attacher container
+		identified in the step 13, where CSI Attacher is the leader.
+		16. Wait until the POD count goes down to 5
+		17. Scale up replica count 20
+		18. PVC's, POD's should come up on all the topology levels specified.
+		19. Verify node affinity details on PV's should be proper with all 5 level details.
+		20. Delete statefulset
+		21. Delete PVC's and SC
+	*/
+
+	ginkgo.It("Volume provisioning when multiple statefulsets creation in progress and in "+
+		"between zones hosts and container nodes are down", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		sts_count = 3
+		statefulSetReplicaCount = 10
+		var ssPods *v1.PodList
+		containerName := "csi-provisioner"
+
+		// get cluster details
+		clusterComputeResource, _, err = getClusterName(ctx, &e2eVSphere)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		/* Get current leader Csi-Controller-Pod where CSI Provisioner is running and " +
+		find the master node IP where this Csi-Controller-Pod is running */
+		ginkgo.By("Get current leader Csi-Controller-Pod name where CSI Provisioner is running and " +
+			"find the master node IP where this container leader is running")
+		controller_name, k8sMasterIP, err := getK8sMasterNodeIPWhereContainerLeaderIsRunning(ctx,
+			client, sshClientConfig, containerName)
+		framework.Logf("CSI-Provisioner is running on Leader Pod %s "+
+			"which is running on master node %s", controller_name, k8sMasterIP)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Get allowed topologies for Storage Class
+		allowedTopologyForSC := getTopologySelector(topologyAffinityDetails, topologyCategories,
+			topologyLength)
+
+		// Create SC with WFC BindingMode
+		ginkgo.By("Creating Storage Class with WFC Binding Mode and allowed topolgies of 5 levels")
+		storageclass, err := createStorageClass(client, nil, allowedTopologyForSC, "",
+			bindingMode, false, "nginx-sc")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Bring down all esxi hosts that belongs to zone2
+		ginkgo.By("Bring down all ESXi hosts that belongs to zone2")
+		hostsInCluster := getHostsByClusterName(ctx, clusterComputeResource, topologyClusterList[1])
+		powerOffHostsList = powerOffEsxiHostByCluster(ctx, &e2eVSphere, topologyClusterList[1],
+			len(hostsInCluster))
+		defer func() {
+			ginkgo.By("Bring up all ESXi host which were powered off in zone2")
+			for i := 0; i < len(powerOffHostsList); i++ {
+				powerOnEsxiHostByCluster(powerOffHostsList[i])
+			}
+			k8sMasterIPs := getK8sMasterIPs(ctx, client)
+			checkNodesStatus := "kubectl get nodes | grep NotReady |  awk '{print $1}'"
```

Comment:
Can we check if nodes are in NotReady state through API instead of running a kubectl command

Reply:
ack